proxy/envoy/discovery_test.go

```diff
@@ -22,29 +22,56 @@ import (
 	"testing"
 
 	restful "github.com/emicklei/go-restful"
+	"github.com/golang/protobuf/proto"
 
-	proxyconfig "istio.io/api/proxy/v1/config"
 	"istio.io/manager/model"
 	"istio.io/manager/test/mock"
 	"istio.io/manager/test/util"
 )
 
-func makeDiscoveryService(r *model.IstioRegistry) *DiscoveryService {
-	return &DiscoveryService{
-		services: mock.Discovery,
-		config:   r,
-		mesh:     &DefaultMeshConfig,
+type mockController struct{}
```

Comment:
I intended mockController to be the minimal code that satisfied the expected interface for testing discovery service. As such, I didn't want to implement unneeded mock features which would need to be maintained if they weren't be used. If we need a full featured test version of the controller it might be better to implement a more general purpose`fakeController` that behaves like the real controller that can be used by any component.